Wikipedia/Code/PageHistoryViewController.swift

```diff
@@ -182,6 +189,28 @@ class PageHistoryViewController: ColumnarCollectionViewController {
         getPageHistory()
     }
 
+    override func viewDidAppear(_ animated: Bool) {
+        super.viewDidAppear(animated)
+        guard let initialRevID = revisionToScrollTo else {
```

Comment:
@mcleinman I think you'll also need to move this logic to run in the completion of the `getPageHistory` fetch, otherwise the data might not be there yet by the time you seek out this cell.

Reply:
> > "View changes" buttons work, unless you're far enough back on the AALD VC that the given event wouldn't initially be loaded when opening the history VC. In that case, it just loads the history screen. Do we think this is okay? Or should we add even more logic to the history VC to allow for this case?
> 
> @mcleinman I think we should try to allow for this case - the revisions API that the history screen hits has an easy way to fetch results up to a certain revision number which we can hopefully lean on. Try adding an `rvendid` parameter in the `fetchRevisionInfo` [method](https://github.com/wikimedia/wikipedia-ios/blob/main/Wikipedia/Code/PageHistoryFetcher.swift#L8) and removing `rvlimit` for this first fetch if they're coming from the View Changes button. I'm hoping this will work better than progressively paging until we find that revision.
> 
> Some extra info from my digging:
> [Here's](https://www.mediawiki.org/wiki/API:Revisions) the API documentation and [here's](https://en.wikipedia.org/w/api.php?action=query&prop=revisions&titles=Main%20Page&rvprop=timestamp%7Cids%7Cuser%7Cparsedcomment%7Csize&rvendid=956737310) an example of fetching Main_Page up to revision #956737310.

I have this working... except if the snippet is one of the first few, we have a very limited history page shown initially. Ideally we'd have a `minNumberOfRevisions` parameter, so it would return the maximum of "all revisions through `revEndId`" and "`revLimit`". If we think people are going to tap items in the first 51 revisions more often than ones below that, we should probably keep the behavior here as is.

Reply:
@mcleinman hmm okay, can you take a video of how it looks and send to Carolyn? Maybe she has thoughts on ways to improve with a loading state or something. Another thought is we could try fetching the first page of history data as soon as they land on the AaaLD modal, so that when they tap view changes the data is already there (while it's fetching more up to the `revEndId`. I'm not sure how much refactoring that would require though. If `URLCache` will kick in and give you back something you just fetched a few minutes ago, maybe it would require very little. Just a thought though, I think we leave the behavior as-is for now and revisit at the end once we are feature complete with the thank flow. Thanks for looking into it!

Reply:
I may have a better solution - working on it right now.

Reply:
Should be working better now - can handle scrolling to the proper spot, no matter what. Had some issues using `revEndId` because `batchComplete` was off, and so we couldn't progressively load any more items as we scrolled. (And because batch wasn't complete, we weren't actually showing the element we wanted.) For simplicity, we're now just hitting the server until we get the full load which includes the target history item.